mne/surface.py

```diff
@@ -1917,6 +1917,14 @@ def _marching_cubes(image, level, smooth=0, fill_hole_size=None, use_flying_edge
             f"{level.size} elements"
         )
 
+    # vtkImageData indexes as slice, row, col (Z, Y, X):
+    # https://discourse.vtk.org/t/very-confused-about-imdata-matrix-index-order/6608/2
+    # We can accomplish his by requiring F continuity and raveling later.
+    # We also pass double as passing integer types directly can be problematic!
+    image = np.asfortranarray(image, dtype=float)
+    image_shape = image.shape
+    assert image.flags.f_contiguous
```

Comment:
@alexrockhill feel free to look and see if this makes sense

Reply:
Yes, that makes sense, I assume it works. I wasn't able to figure out what SetScalars was expecting the matrix order to be, much better job Googling than me. Seems like a good solution.